lib/converter.js

```diff
@@ -70,9 +70,43 @@ function convert(url, options) {
             return;
         }
 
-        var result = "data:image/png;base64," + page.renderBase64("PNG");
-        system.stdout.write(result);
-        phantom.exit();
+        /*
+         * We want to wati until the standard deviation of the previous 100 FPS checks
+         * is less than 1.5, to ensure the svg has rendered properly. The render rate
+         * tends to fluctuate while rendering the svg, and level out when it is finished.
+        */
+        page.evaluate(function() {
+            var previousT = 0;
+            var t = 0;
+            var fpsList = [];
+
+            function stdev(values) {
+                var mean = values.reduce(function(a, b) { return a + b; }) / values.length;
+                var devSquared = values.map(function(a) { return Math.pow(a - mean, 2); }).reduce(function(a, b) { return a + b; });
+                return Math.sqrt(devSquared / (values.length - 1));
+            }
+
+            function calcFPS(now) {
+                t = now;
+                var fps = 1000 / (t - previousT);
+                fpsList.push(fps);
+                var std = stdev(fpsList.slice(fpsList.length - 100, fpsList.length));
+                if (std <= 1.5) {
+                    console.log("done");
+                    return;
+                }
+                previousT = t;
+                requestAnimationFrame(calcFPS);
+            }
+
+            requestAnimationFrame(function() { console.log("done"); });
+        });
+
+        page.onConsoleMessage = function(msg) {
```

Comment:
this is absolutely hilarious to me, can i suggest we at least check if the console is a done msg?

Reply:
yeah had this in the original code but it got moved around so much while testing it.